components/table/stories/template.js

```diff
@@ -101,59 +138,73 @@ export const TableRowItem = ({
-									customClasses: [`${rootClass}-disclosureIcon`],
-									ariaExpanded: isExpanded,
-									ariaControls,
-								}, context)
-							)}
-							${useThumbnail ? getCellContent(0) : html`<div class="${rootClass}-collapseContent">${getCellContent(0)}</div>`}
-						</div>
-					</${cellTag}>`
-				: html`
-					<${cellTag}
-						role=${ifDefined(showCheckbox ? "gridcell" : useDivs ? "cell" : undefined)}
-						class=${classMap({
-							[`${rootClass}-cell`]: true,
-							[`${rootClass}-cell--thumbnail`]: useThumbnail,
-							[`${rootClass}-cell--divider`]: useColumnDividers,
-						})}
-						colspan=${ifDefined(isSectionHeader && showCheckbox ? "4" : isSectionHeader ? "3" : undefined)}
-					>${getCellContent(0)}</${cellTag}>`
+			? html`
+				<${cellTag}
+					role=${ifDefined(showCheckbox ? "gridcell" : useDivs ? "cell" : undefined)}
+					class=${classMap({
+						[`${rootClass}-cell`]: true,
+						[`${rootClass}-cell--collapsible`]: true,
+						[`${rootClass}-cell--visual`]: useVisuals,
+						[`${rootClass}-cell--divider`]: useColumnDividers,
+						[`${rootClass}-cell--alignEnd`]: getTextAlignment(0) === "end",
+						...cellCustomClasses?.[showCheckbox ? 1 : 0]?.reduce((a, c) => ({ ...a, [c]: true }), {}),
+					})}
+					tabindex="0"
+				>
+					<div class="${rootClass}-collapseInner">
+						${when(!isLastTier, () =>
+							Button({
+								size: "m",
+								iconName: "ChevronRight100",
+								iconSet: "ui",
+								hideLabel: true,
+								customClasses: [`${rootClass}-disclosureIcon`],
+								ariaExpanded: isExpanded,
+								ariaControls,
+							}, context)
+						)}
+						${useVisuals ? getCellContent(0) : html`<div class="${rootClass}-collapseContent">${getCellContent(0)}</div>`}
+					</div>
+				</${cellTag}>`
+			: html`
+				<${cellTag}
+					role=${ifDefined(showCheckbox ? "gridcell" : useDivs ? "cell" : undefined)}
+					class=${classMap({
+						[`${rootClass}-cell`]: true,
+						[`${rootClass}-cell--visual`]: useVisuals,
+						[`${rootClass}-cell--divider`]: useColumnDividers,
+						[`${rootClass}-cell--alignEnd`]: getTextAlignment(0) === "end",
+						...cellCustomClasses?.[showCheckbox ? 1 : 0]?.reduce((a, c) => ({ ...a, [c]: true }), {}),
+					})}
+					colspan=${ifDefined(isSectionHeader && showCheckbox ? "4" : isSectionHeader ? "3" : undefined)}
+					tabindex="0"
+				>
+					${getCellContent(0)}
+				</${cellTag}>`
 		}
 
 		${when(!isSectionHeader, () => html`
 			<${cellTag}
 				role=${ifDefined(showCheckbox ? "gridcell" : useDivs ? "cell" : undefined)}
 				class=${classMap({
 					[`${rootClass}-cell`]: true,
-					[`${rootClass}-cell--thumbnail`]: useThumbnail,
+					[`${rootClass}-cell--visual`]: useVisuals,
 					[`${rootClass}-cell--divider`]: useColumnDividers,
+					[`${rootClass}-cell--alignEnd`]: getTextAlignment(1) === "end",
+					...cellCustomClasses?.[showCheckbox ? 2 : 1]?.reduce((a, c) => ({ ...a, [c]: true }), {}),
 				})}
+				tabindex="0"
```

Comment:
The rows should be keyboard focusable, as well as the cells (according to the designs), so I added the tabindex in quite a few places.

Any accessibility concerns with how everything is laid out right now? I don't know the flow for tables. Should each focused row move to the next focusable row, or would a user focus on a row, then move through the cells?

Reply:
Ok, I think that tabindex will go away when you update this to be a button instead (per Slack discussion), since that will provide the tab stop.

Scrolling via keyboard works as I expect 👍